### PR TITLE
My Jetpack: implement first approach of data handling

### PR DESCRIPTION
**Pull request by retrofox**
This PR implements the initial approach for data handling of My Jetpack app. As an implementation real use case, it introduces a custom hook, redux changes, and also backend code to make the communication end to end work.

The most important change introduced here is the side effect handling. PR adds a [Readme file](https://github.com/Automattic/jetpack/blob/00f964889aed067d232e82789367bd0ffea3db20/projects/packages/my-jetpack/_inc/state/README.md) that we can keep documenting.

Fixes #

#### Changes proposed in this Pull Request:

* My Jetpack: implement first approach of data handling

#### Does this pull request change what data or activity we track or use?

no

#### Testing instructions:

I added a simple Button in the dashboard in order to test the communication, end to end.

Screen Shot 2022-01-20 at 3 22 07 PM

That is as ugly as it is useful, just like me 😅 

### Check Actions

* Open the redux tool
* Select `my-jetpack` store
* Click on the button.
* Confirm you see the actions there.

https://user-images.githubusercontent.com/77539/150399727-c08d9d06-5448-4083-995f-c8737dd02f45.mov

### Check side effects
The activate/deactivate actions are observed by the middleware. You can see how the client performs the request:

https://user-images.githubusercontent.com/77539/150400253-174f2bc3-faba-4edd-b32c-9b1930f77387.mov

In short, the actions triggered from the UI are `ACTIVATE_PRODUCT` and `DEACTIVATE_PRODUCT`. Both are caught by the middleware, processed to finally dispatch the `setProductStatus()` action (`SET_PRODUCT_STATUS`) if everything is fine. Otherwise, it will dispatch `setProductActionError()`, storing the error into the state tree. 

For now, the backend responses with hardcoded data.

### CODE REVIEW

**Comment by retrofox** [comment]
close it in favor of https://github.com/Automattic/jetpack/pull/22448
